src/microbe_stage/Microbe.cs

```diff
@@ -712,4 +720,113 @@ private void CountHexes()
 
         cachedHexCountDirty = false;
     }
+
+    /// <summary>
+    ///   Provides extra level of abstraction to allow simultaneous switching between 3D positional and non positional
+    ///   audio players in a single Spatial-derived class.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Useful in cases where the playing of an audio stream must be able to fulfill both of these conditions.
+    ///   </para>
+    /// </remarks>
+    private class AudioPlayer : Spatial
```

Comment:
This should probably be in a new file in the engine folder, as I presume for 3D character stages we might have the same issue as currently.